packages/core/src/ckb/transaction.ts

```diff
@@ -1193,6 +1208,83 @@ export class Transaction extends mol.Entity.Base<
       position,
     };
   }
+  /**
+   * Computes the signing hash information for a given script, specified in the spec:
+   * https://github.com/nervosnetwork/rfcs/pull/446
+   *
+   * @param scriptLike - The script associated with the transaction, represented as a ScriptLike object.
+   * @param client - The client for complete extra infos in the transaction.
+   * @returns A promise that resolves to an object containing the signing message and the witness position,
+   *          or undefined if no matching input is found.
+   *
+   * @example
+   * ```typescript
+   * const signHashInfo = await tx.getTxMessageAll(scriptLike, client);
+   * if (signHashInfo) {
+   *   console.log(signHashInfo.message); // Outputs the signing message
+   *   console.log(signHashInfo.position); // Outputs the witness position
+   * }
+   * ```
+   */
+  async getTxMessageAll(
+    scriptLike: ScriptLike,
+    client: Client,
+    hasher: Hasher = new HasherCkb(),
+  ): Promise<{ message: Hex; position: number } | undefined> {
+    const script = Script.from(scriptLike);
+    let position = -1;
+    hasher.update(this.hash());
+
+    for (let i = 0; i < this.inputs.length; i += 1) {
+      const { cellOutput, outputData } = await this.inputs[i].getCell(client);
+      hasher.update(cellOutput.toBytes());
+      Transaction.hashBytesToHasher(outputData, hasher);
+    }
+
+    for (let i = 0; i < this.witnesses.length; i += 1) {
+      const input = this.inputs[i];
+      if (input) {
+        const { cellOutput } = await input.getCell(client);
+
+        if (!script.eq(cellOutput.lock)) {
+          continue;
+        }
+
+        if (position === -1) {
+          position = i;
+        }
+      }
+
+      if (position === -1) {
+        return undefined;
+      }
+      if (i == position) {
```

Comment:
`===` is almost always more recommended than `==` to avoid implicit conversion.

Reply:
fixed